### PR TITLE
Feature/fillable calc

### PR DESCRIPTION
**Pull request by cryptomental**
##### Description
This PR adds RemainingFillableCalc. Public API is the same as 0x's.

##### Checklist

- [x] Linter status: 100% pass
- [x] Changes don't break existing behavior
- [x] Tests coverage hasn't decreased

##### Affected core subsystem(s)
Order watcher, tests.

##### Testing
Tested with RemainingFillableCalc.test.ts

##### Refers/Fixes
Fixes: #39

### CODE REVIEW

File: src/order_watcher/RemainingFillableCalc.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  }
+
+  private async _hasTakerSufficientFundsForFee(): Promise<boolean> {
+    const takerMktBalance = await this._getAvailableFeeFunds(this._signedOrder.taker);
+    const takerFeeNeeded = this._signedOrder.takerFee;
+
+    return takerMktBalance.gte(takerFeeNeeded);
+  }
+
+  private async _getAvailableFeeFunds(accountAddress: string): Promise<BigNumber> {
```
Comment (pelsasser) [inline src/order_watcher/RemainingFillableCalc.ts]: @cryptomental - I think it makes sense to also add a check here to make sure that they have approved the funds be transferred. There is a similar example in `traderOrderAsync` using `getAllowanceAsync`.  What do you think?

File: test/RemainingFillableCalc.test.ts [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+    await collateralToken.transferTx(takerAddress, initialCredit).send({ from: deploymentAddress });
+    await collateralToken
+      .approveTx(collateralPoolAddress, initialCredit)
+      .send({ from: takerAddress });
+    await depositCollateralAsync(web3.currentProvider, collateralPoolAddress, initialCredit, {
+      from: takerAddress
+    });
+  });
+
+  afterEach(async () => {
```
Comment (pelsasser) [inline test/RemainingFillableCalc.test.ts]: may want to consider using the new `createEVMSnapshot` and `restoreEVMSnapshot` functions here.  Not needed to get this merged, we can always add a ticket to it later.

File: src/order_watcher/RemainingFillableCalc.ts (hunk 1 above)
Comment (cryptomental) [reply src/order_watcher/RemainingFillableCalc.ts]: Makes sense. I will add the checks.

File: test/RemainingFillableCalc.test.ts (hunk 2 above)
Comment (cryptomental) [reply test/RemainingFillableCalc.test.ts]: Ok, I will try to add this for this PR as well.

File: src/order_watcher/RemainingFillableCalc.ts (hunk 1 above)
Comment (cryptomental) [reply src/order_watcher/RemainingFillableCalc.ts]: Done. Checks added.

File: test/RemainingFillableCalc.test.ts (hunk 2 above)
Comment (cryptomental) [reply test/RemainingFillableCalc.test.ts]: EVMSnapshot added as well.

Comment (pelsasser) [reply test/RemainingFillableCalc.test.ts]: do we need the withdraw functions still if we are reverting state back to before he deposits?

Comment (cryptomental) [reply test/RemainingFillableCalc.test.ts]: Good point. I amended the last commit and removed the the withdraw functions.

Comment (pelsasser) [reply test/RemainingFillableCalc.test.ts]: LGTM!

**Pull request merged**